### PR TITLE
New methods in interfaces is a BC-break

### PR DESCRIPTION
**Pull request by joshdifabio**
Adding a method to an interface breaks any existing implementation of that interface. Shouldn't doing this be considered a BC-break?

### CODE REVIEW

**Comment by markstory** [comment]
I half agree. Removing the option to add methods means we would have to create new interfaces that extend the old ones. This could also break compatibility in other ways.

**Comment by joshdifabio** [comment]
That's the [open/closed principle](http://en.wikipedia.org/wiki/Open/closed_principle): open to extension, closed to modification. In my humble opinion, extending an interface in order to extend its functionality is perfectly reasonable and correct.

**Comment by ravage84** [comment]
Even though I tend to agree, I see this limiting our options.
How often would that actually happen?
How often did that happen in the past?

I would rephrase it with some kind of middle way stating that we "normally" don't add methods to existing interfaces or something like that...

**Comment by markstory** [comment]
@ravage84 It has happened in the past a few times. In those cases we didn't add the methods to the interface, and instead used `method_exists()` calls to check the implementing class.

**Comment by markstory** [comment]
Merging as new interface methods do cause a non-zero amount of upgrading grief.
